R/loadFromMothur.R

```diff
@@ -0,0 +1,206 @@
+#'   path of the taxonomy table to be imported. (default:
+#'   \code{taxonomyTableFile = NULL}).
+#'
+#' @param sampleMetaFile a single \code{character} value defining the file path
+#'   of the sample metadata to be imported. (default: \code{sampleMetaFile = NULL}).
+#'
+#' @param featureNamesAsRefSeq \code{TRUE} or \code{FALSE}: Should the feature
+#'   names of the feature table be regarded as reference sequences? This setting
+#'   will be disregarded, if \code{refSeqFile} is not \code{NULL}. If the
+#'   feature names do not contain valid DNA characters only, the reference
+#'   sequences will not be set.
+#'
+#' @param refSeqFile a single \code{character} value defining the file path of
+#'   the reference sequences for each feature. (default: \code{refSeqFile =
+#'   NULL}).
+#'
+#' @param phyTreeFile a single \code{character} value defining the file path of
+#'   the phylogenetic tree. (default: \code{phyTreeFile = NULL}).
+#'
+#' @param ... additional arguments:
+#'
+#' @details
+#' Results exported from Mothur can be imported as a
+#' \code{TreeSummarizedExperiment} using \code{loadFromMothur}. Except for the
+#' \code{featureTableFile}, the other data types, \code{taxonomyTableFile},
+#' \code{refSeqFile} and \code{phyTreeFile}, are optional, but are highly
+#' encouraged to be provided.
+#'
+#' @return  A
+#' \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#' object
+#'
+#' @name loadFromMothur
+#' @seealso
+#' \code{\link[=makeTreeSummarizedExperimentFromphyloseq]{makeTreeSummarizedExperimentFromphyloseq}}
+#' \code{\link[=makeTreeSummarizedExperimentFromBiom]{makeTreeSummarizedExperimentFromBiom}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{makeTreeSummarizedExperimentFromDADA2}}
+#' \code{\link[=makeTreeSummarizedExperimentFromDADA2]{loadFromQIIME2}}
+#'
+#' @export
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @references
+#' \url{https://mothur.org/}
+#' 
+#' @examples
+#' # Abundance table
+#' counts <- "~/Desktop/Baxter_FITs_Microbiome_2016_fit.final.tx.1.subsample.shared"
+#' # Taxa table
+#' taxa <- "~/Desktop/Baxter_FITs_Microbiome_2016_fit.final.tx.1.cons.taxonomy"
+#' # Sample meta data
+#' meta <- "~/Desktop/Baxter_FITs_Microbiome_2016_mapping.csv"
+#' 
+#' # Creates tse object from files
+#' \dontrun{tse <- loadFromMothur(counts, taxa, meta)}
+#' 
+
+loadFromMothur <- function(featureTableFile,
+                           taxonomyTableFile = NULL,
+                           sampleMetaFile = NULL,
+                           featureNamesAsRefSeq = TRUE,
+                           refSeqFile = NULL,
+                           phyTreeFile = NULL,
+                           ...) {
+  
+    # input check
+    if(!.is_non_empty_string(featureTableFile)){
+      stop("'featureTableFile' must be a single character value.",
+           call. = FALSE)
```

Comment:
Are the indentations here and below correct? Shouldn't it always be multiples of 4 by Bioc standards?

Reply:
yep those are not correct.

Reply:
Fixed